lib/evss/service.rb

```diff
@@ -6,18 +6,23 @@
 module EVSS
   class Service < Common::Client::Base
     STATSD_KEY_PREFIX = 'api.evss'
+    SYSTEM_NAME = 'vets.gov'
 
     def initialize(user)
       @user = user
     end
 
-    def perform(method, path, body = nil, headers = {})
+    def perform(method, path, body = nil, headers = {}, &block)
       headers = headers_for_user(@user).merge(headers)
-      super(method, path, body, headers)
+      super(method, path, body, headers, &block)
     end
 
     private
 
+    def perform_json(method: :post, path:, body: nil, headers: {})
```

Comment:
I think [FaradayMiddleware::EncodeJson](https://github.com/lostisland/faraday_middleware/blob/master/lib/faraday_middleware/request/encode_json.rb) will encode the body and set the Content-Type for you if it's added the configuration for the service. Appeals service [uses that middleware](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/appeals/configuration.rb#L28). Although I'm not sure how that would affect the `all_claims` endpoint, it might work or pass through unaffected, but if not what you have here may be the way to go.

Reply:
@kreek thanks, i changed it